File: amic-platform/src/types/notification.ts
export type NotificationModule = "fdd" | "kiis" | "im" | "portal";

export interface NotificationItem {
  id: string;
  module: NotificationModule;
  type: string;
  title: string;
  message: string;
  is_read: boolean;
  created_at: string;
  link: string | null;
}

File: fdd/backend/app/schemas/notification.py
"""Notification Pydantic schemas — Phase 5."""

import uuid
from datetime import datetime

from pydantic import BaseModel

from app.models.notification import NotificationModule


class NotificationRead(BaseModel):
    id: uuid.UUID
    module: NotificationModule
    type: str
    title: str
    message: str
    is_read: bool
    created_at: datetime
    link: str | None

    model_config = {"from_attributes": True}


class NotificationCreate(BaseModel):
    """Internal-use schema for creating notifications programmatically."""

    user_id: uuid.UUID
    module: NotificationModule
    type: str
    title: str
    message: str
    link: str | None = None
